src/hotspot/cpu/arm/stubGenerator_arm.cpp

```diff
@@ -3029,11 +3097,13 @@ class StubGenerator: public StubCodeGenerator {
 
 
  public:
-  StubGenerator(CodeBuffer* code, bool all) : StubCodeGenerator(code) {
-    if (all) {
-      generate_all();
-    } else {
+  StubGenerator(CodeBuffer* code, int phase) : StubCodeGenerator(code) {
+    if (phase == 0) {
       generate_initial();
+    } else if (phase == 1) {
+      generate_phase1();
+    } else {
+      generate_all();
```

Comment:
It might be better to check for phases explicitly:
  
     } else if (phase == 2) {
       generate_all();
     } else {
        ShouldNotReachHere();
     }

Reply:
I agree, but it matches what all other ports do :)